docs/src/api.md

```diff
@@ -1,26 +1,80 @@
-# AdvancedHMC.jl
+# Detailed API for AdvancedHMC.jl
 
-Documentation for AdvancedHMC.jl
+An important design goal of AdvancedHMC.jl is modularity; we would like to support algorithmic research on HMC.
+This modularity means that different HMC variants can be easily constructed by composing various components, such as preconditioning metric (i.e., mass matrix), leapfrog integrators, trajectories (static or dynamic), adaption schemes, etc. In this documentation, we will explain the detailed usage of different modules in AdancedHMC.jl to provide a comprehensive udnerstanding of how AdvancedHMC.jl can achieve both modularity and efficiency.
 
-```@contents
-```
+### [Hamiltonian mass matrix (`metric`)](@id hamiltonian_mm)
```

Comment:
Could we add a sentence before this section saying that we first list some of the exported types/features by categories, but that full docs can be found below? It took me a while to realise that the full ```@docs``` entries of all these are below. Alternatively, we could put in ```@docs``` entries here for all the ones we want to highlight category-by-category, and leave the rest for the ```@autodocs``` block. That way there would also be less to maintain (e.g. if the constructor arguments for one of these change).